composer/trainer/trainer.py

```diff
@@ -1020,9 +1099,248 @@ def saved_checkpoints(self) -> List[Tuple[Timestamp, List[pathlib.Path]]]:
+            assert trainer.state.timestamp.epoch == "3ep"
+
+        Args:
+            train_dataloader (Iterable | DataSpec | Dict[str, Any], optional): See :class:`.Trainer`.
+            train_dataloader_label (str, optional): See :class:`.Trainer`.
+            train_subset_num_batches (int, optional): See :class:`.Trainer`.
+            compute_training_metrics (bool, optional): See :class:`.Trainer`.
+            reset_time (bool): Whether to reset the :attr:`.State.timestamp` to zero values. Defaults to False.
+
+                If ``True``, the timestamp will be zeroed out, causing :class:`.ComposerScheduler` and
+                :class:`.Algorithm` instances to start from the beginning, as if it is a new training run. The model
+                will be trained for ``duration``, if specified, or for :attr:`.State.max_duration`, which would have
+                been provided when construting the :class:`.Trainer` or by a previous call to :meth:`.fit`.
+
+                .. note::
+
+                    Model gradients, optimizer states, and native PyTorch schedulers will not be reset.
+
+                If ``False`` (the default), training time will be incremented from where the previous call to
+                :meth:`.fit` finished (or from zero, if a new training run).
+                The :attr:`~.State.max_duration` will be incremented by the ``duration`` parameter.
+
+            duration (Time[int] | str | int, optional): The duration to train. Can be an integer, which will be
+                interpreted to be epochs, a str (e.g. ``1ep``, or ``10ba``), or a :class:`.Time` object.
+
+                If ``reset_time`` is False (the default), then :attr:`.State.max_duration` will be converted
+                into the same units as this parameter (if necessary), and then the max duration incremented by the
+                value of this parameter.
+
+                If ``reset_time`` is True, then :attr:`.State.max_duration` will be set to this parameter.
+
+            optimizers (torch.optim.Optimizer | Sequence[torch.optim.Optimizer], optional): See :class:`.Trainer`.
+            schedulers (PyTorchScheduler | ComposerScheduler | Sequence[PyTorchScheduler | ComposerScheduler], optional):
+                See :class:`.Trainer`.
+            scale_schedule_ratio (float, optional): See :class:`.Trainer`.
+            step_schedulers_every_batch (bool, optional): See :class:`.Trainer`.
+            eval_dataloader (Iterable | DataSpec | Evaluator | Sequence[Evaluator], optional): See :class:`.Trainer`.
+            eval_subset_num_batches (int, optional): See :class:`.Trainer`.
+            eval_interval (int | str | Time | (State, Event) -> bool, optional): See :class:`.Trainer`.
+            grad_accum (int | str, optional): See :class:`.Trainer`.
+            precision (Precision | str, optional): See :class:`.Trainer`.
+            grad_clip_norm (float, optional): See :class:`.Trainer`.
```

Comment:
Maybe add a comment here that setting this value in fit if you are using deepspeed is not allowed?

Reply:
Good point; added a comment.